logging/impl.go

```diff
@@ -71,24 +84,42 @@ func (impl *impl) Sync() error {
 	return multierr.Combine(errs...)
 }
 
-func (impl *impl) With(args ...interface{}) *zap.SugaredLogger {
-	return nil
+func (imp *impl) With(args ...interface{}) *zap.SugaredLogger {
+	return imp.AsZap().With(args...)
 }
 
-func (impl *impl) WithOptions(opts ...zap.Option) *zap.SugaredLogger {
-	return nil
+func (imp *impl) WithOptions(opts ...zap.Option) *zap.SugaredLogger {
+	return imp.AsZap().WithOptions(opts...)
 }
 
-func (impl *impl) AsZap() *zap.SugaredLogger {
-	return NewLogger("").AsZap()
+func (imp *impl) AsZap() *zap.SugaredLogger {
+	// A hack to primarily copy zap's observer core objects from the viam logger back down to the
```

Comment:
A change worthy of understanding. Hopefully the comment here explains what's going on. But when reading the code-review top to bottom, this won't make much sense. One has to first look at and understand [our testing log logic](https://github.com/viamrobotics/rdk/pull/3185/files#diff-6bb8ce03ad8192446d441a3d85bf0fb8307e3d7829bc027785b35ce95868da70L79-R83) that uses an "observed logger".

Reply:
This is a great observation. The intention of this code is that it was only necessary for testing.

But this has made me realize that's not good enough. The way the code is right now, we lose the net logger (send logs to `app`) when calling this. I need to do something about that.